diff --git a/src/crc32c_arm64.cc b/src/crc32c_arm64.cc
index 701326d..e5047d2 100644
--- a/src/crc32c_arm64.cc
+++ b/src/crc32c_arm64.cc
@@ -78,7 +78,7 @@ uint32_t ExtendArm64(uint32_t crc, const uint8_t *data, size_t size) {
     crc3 = 0;
 
     // Process 1024 bytes in parallel.
-    CRC32C1024BYTES(data);
+    CRC32C32BYTES(data, 0);
 
     // Merge the 4 partial CRC32C values.
     t2 = (uint64_t)vmull_p64(crc2, k2);
